### PR TITLE
Prot warrior update

### PR DESCRIPTION
**Pull request by JooperGH**
- Updated all of the talents in the protection tree, except for the block part of Critical Block. (Its tricky)
- Updated gear presets & talent presets.
- Implemented new skills & effects.
- Implemented some glyphs.
- Deleted old TBC talents.

### CODE REVIEW

File: sim/warrior/devastate.go [hunk 1]
```diff
@@ -53,6 +89,9 @@ func (warrior *Warrior) registerDevastateSpell() {
 				if spellEffect.Landed() {
 					if !warrior.ExposeArmorAura.IsActive() {
 						warrior.SunderArmorDevastate.Cast(sim, spellEffect.Target)
+						if hasGlyph {
```
Comment (Paur-SW) [inline sim/warrior/devastate.go:92]: I might be wrong, but isn't this handled in sunder_armor.go lines 36 and 57? Could be duplicate logic here.

File: sim/warrior/revenge.go [hunk 2]
```diff
@@ -4,10 +4,27 @@ import (
 	"time"
 
 	"github.com/wowsims/wotlk/sim/core"
+	"github.com/wowsims/wotlk/sim/core/proto"
 	"github.com/wowsims/wotlk/sim/core/stats"
 )
 
 func (warrior *Warrior) registerRevengeSpell(cdTimer *core.Timer) {
+	actionID := core.ActionID{SpellID: 30357}
+	warrior.revengeProcAura = warrior.RegisterAura(core.Aura{
```
Comment (Paur-SW) [inline sim/warrior/revenge.go:13]: Nice, much cleaner this way imo.

File: sim/warrior/devastate.go (hunk 1 above)
Comment (JooperGH) [reply sim/warrior/devastate.go:92]: Oh! You're absolutely right, thank you!

File: sim/warrior/shouts.go [hunk 3]
```diff
@@ -33,16 +33,17 @@ func (warrior *Warrior) makeShoutSpellHelper(actionID core.ActionID) *core.Spell
...
-			warrior.shoutExpiresAt = sim.CurrentTime + warrior.shoutDuration
+			warrior.shoutExpiresAt = sim.CurrentTime + warrior.shoutDuration + extraDuration
 		},
 	})
 }
 
 func (warrior *Warrior) makeShoutSpell() *core.Spell {
 	if warrior.ShoutType == proto.WarriorShout_WarriorShoutBattle {
-		return warrior.makeShoutSpellHelper(core.ActionID{SpellID: 47436})
+		return warrior.makeShoutSpellHelper(core.ActionID{SpellID: 47436}, 0)
```
Comment (Paur-SW) [inline sim/warrior/shouts.go:43]: No Glyph of Battle? Is that handled elsewhere?

**Pull request closed without merging**